domains/Links/Policies/LinkPolicy.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Domains\Links\Policies;
+
+use Domains\Accounts\Enums\AccountTypeEnum;
+use Domains\Accounts\Models\User;
+use Domains\Links\Models\Link;
+use Illuminate\Auth\Access\HandlesAuthorization;
+
+class LinkPolicy
+{
+    use HandlesAuthorization;
+
+    public function create(?User $user, string $author_email)
+    {
+        $pendingCount = Link::forAuthorWithEmail($author_email)
+            ->unapproved()
+            ->count();
+
+        return optional($user)->hasRole(AccountTypeEnum::EDITOR)
+            || optional($user)->isTrusted()
+            || $pendingCount < config('links.max_unapproved_links');
```

Comment:
We could improve the performance of this policy by returning an early true value if the `$user` is not `null` and is `trusted` or has, at least, the `EDITOR` role, before allowing the system to execute the query to check for how many links are already registered and unapproved.